science/mpich/Portfile

```diff
@@ -86,6 +64,7 @@ array set clist {
     gcc6    {macports-gcc-6}
     gcc7    {macports-gcc-7}
     gcc8    {macports-gcc-8}
+    devel   {clang} # Here to prevent portindex failure on -devel
```

Comment:
This isn't the right fix. Among other things, this causes you to declare a subport called `mpich-devel-devel`. Instead, don't run the code that uses `clist` when in one of the obsolete subports.

Reply:
Done